platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/MemoryIndexDiskKeyValueStore.java

```diff
@@ -62,18 +46,18 @@
  * @param <D> type for data items
  */
 @SuppressWarnings({"DuplicatedCode"})
-public class MemoryIndexDiskKeyValueStore<D> implements AutoCloseable, Snapshotable {
+public class MemoryIndexDiskKeyValueStore<D> implements AutoCloseable, Snapshotable, Compactible {
     private static final Logger logger = LogManager.getLogger(MemoryIndexDiskKeyValueStore.class);
 
-    /** This is useful for debugging and validating but is too expensive to enable in production. */
-    protected static boolean enableDeepValidation = logger.isTraceEnabled();
     /**
      * Index mapping, it uses our key as the index within the list and the value is the dataLocation
      * in fileCollection where the key/value pair is stored.
      */
     private final LongList index;
     /** On disk set of DataFiles that contain our key/value pairs */
-    private final DataFileCollection<D> fileCollection;
+    final DataFileCollection<D> fileCollection;
```

Comment:
Is it leaked to outside of this class? Why isn't it kept private?

Reply:
It's required for testing. The reason is rather silly - the minimal number of files to compact is defined in a `config` object which is immutable and resides in the static context. So, in this method `MemoryIndexDiskKeyValueStoreTest#createDataAndCheckImpl` I have to do extend `DataFileComactor` this way

```
     DataFileCompactor createFileCompactor() {
                        return new DataFileCompactor(fileCollection) {
                            @Override
                            int getMinNumberOfFilesToCompact() {
                                return 1;
                            }
                        };
                    }
```
And the only real reason is to get to `getMinNumberOfFilesToCompact`. Making this field package-private is the least evil I can think of